### PR TITLE
[ID-34]Remove user data

### LINKED ISSUE #34: [FEATURE] Remove user data
**Is your feature request related to a problem? Please describe.**
Remove user data

### PR DESCRIPTION
**Pull request by stefanvit**
## Description
Remove user data
**Resolves #34 
## Review Time Estimate
- [ ] Immediately
- [ ] Within a week
- [X] When possible

## Type of changes
- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Other (any another change that does not fall in one of the above categories.)

## Checklist:
- [x] I have signed the Rokwire Contributor License Agreement ([CLA](https://rokwire.org/rokwire_cla)). (_Any contributor who is not an employee of the University of Illinois whose official duties include contributing to the Rokwire software, or who is not paid by the Rokwire project, needs to sign the CLA before their contribution can be accepted._)
- [x] I have updated the [CHANGELOG](../CHANGELOG.md).
- [x] I have read the [Contributor Guidelines](../CONTRIBUTING.md).
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] My change requires updating the documentation.
- [ ] I have made necessary changes to the documentation.
- [ ] I have added tests related to my changes.
- [ ] My changes generate no new warnings.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Comment by stefanvit** [comment]
Hi @petyos, the reason for the error on the build is that i couldn't generate the mocks for some reason. I use the way that is written in the convensions but it hits the wall every time. Could you please help me? Thank you.

**Comment by petyos** [comment]
Hi @stefanvit ,

Please try:
cd core
mockery --name=Storage

File: core/interfaces.go [hunk 1]
```diff
@@ -28,3 +28,9 @@ type Shared interface {
 	isEventAdmin(orgID string, appID string, eventID string, userID string, externalIDs map[string]string) (bool, error)
 	hasAttendedEvent(orgID string, appID string, eventID string, userID string, externalIDs map[string]string) (bool, error)
 }
+
+// Core exposes Core APIs for the driver adapters
+type Core interface {
+	RetrieveCoreUserAccountByCriteria(accountCriteria map[string]interface{}, appID *string, orgID *string) ([]model.CoreAccount, error)
```
Comment (petyos) [inline core/interfaces.go:34]: I fixed the same for Polls BB. No need to add code which is not used. Please look at Polls BB for reference.

**Review by petyos — CHANGES_REQUESTED** [review]
Please do not mock all files but only Storage as it is now. This is out of the issue scope.

File: core/application.go [hunk 2]
```diff
@@ -17,8 +17,9 @@ package core
 import (
 	"application/core/interfaces"
 	"application/core/model"
+	corebb "application/driven/core"
 
-	"github.com/rokwire/core-auth-library-go/v3/authutils"
```
Comment (petyos) [inline core/application.go:21]: Do not downgrade.

File: core/application_test.go [hunk 3]
```diff
@@ -26,7 +26,7 @@ import (
 
 	"github.com/stretchr/testify/mock"
 
-	"github.com/rokwire/core-auth-library-go/v3/authutils"
```
Comment (petyos) [inline core/application_test.go:29]: Do not downgrade.

File: core/model/surveys.go [hunk 4]
```diff
@@ -152,3 +152,28 @@ type OptionData struct {
...
+}
+
+// DeletedMembership defines model for DeletedMembership.
+type DeletedMembership struct {
+	AccountID string                  `json:"account_id"`
+	Context   *map[string]interface{} `json:"context,omitempty"`
+}
+
+// CoreAccount represents an account in the Core BB
+type CoreAccount struct {
```
Comment (petyos) [inline core/model/surveys.go:170]: CoreAccount & CoreProfile are not needed.

File: driven/storage/adapter.go [hunk 5]
```diff
@@ -251,6 +251,36 @@ func (a *Adapter) DeleteConfig(id string) error {
 	return nil
 }
 
+// DeleteSurveyResponsesWithIDs Deletes survey responses
```
Comment (petyos) [inline driven/storage/adapter.go:254]: Are you sure creator_id is the correct filter for both functions?

File: main.go [hunk 6]
```diff
@@ -67,6 +68,8 @@ func main() {
 	// Service registration
 	baseURL := envLoader.GetAndLogEnvVar(envPrefix+"BASE_URL", true, false)
 	coreBBBaseURL := envLoader.GetAndLogEnvVar(envPrefix+"CORE_BB_BASE_URL", true, false)
+	appID := envLoader.GetAndLogEnvVar(envPrefix+"APP_ID", true, false)
```
Comment (petyos) [inline main.go:71]: What is this? You do not have to add app id and org id.

**Review by petyos — CHANGES_REQUESTED** [review]
Please look into the feedback comments.

File: driven/storage/adapter.go [hunk 7]
```diff
@@ -251,6 +251,32 @@ func (a *Adapter) DeleteConfig(id string) error {
...
+func (a Adapter) DeleteSurveyResponsesWithIDs(accountsIDs []string) error {
+	filter := bson.D{
```
Comment (petyos) [inline driven/storage/adapter.go:256]: You do not you filter by app id and org id?

File: driven/storage/adapter.go [hunk 8]
```diff
@@ -251,6 +251,32 @@ func (a *Adapter) DeleteConfig(id string) error {
...
+	_, err := a.db.surveyResponses.DeleteMany(nil, filter, nil)
+	if err != nil {
+		return errors.WrapErrorAction(logutils.ActionDelete, "user", nil, err)
+	}
+	return nil
+}
+
+// DeleteSurveysWithIDs Deletes surveys
+func (a Adapter) DeleteSurveysWithIDs(accountsIDs []string) error {
+	filter := bson.D{
```
Comment (petyos) [inline driven/storage/adapter.go:269]: You do not you filter by app id and org id?

**Review by petyos — COMMENTED** [review]
Two minor comments.

**Pull request merged**